### PR TITLE
Renaming and regeneration

### PR DESCRIPTION
**Pull request by jskeet**
This PR affects:

- Naming of almost all APIs
- Page-streaming
- Resource names

It's split by "one commit per API" but I wouldn't expect each commit to build, as there are interlocking changes too. It's designed more for review than anything else.

### CODE REVIEW

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs [hunk 1]
```diff
@@ -155,18 +147,14 @@ public void CreateSubscription()
             string topicId = _fixture.CreateTopicId();
             string subscriptionId = _fixture.CreateSubscriptionId();
 
-            PublisherClient.Create().CreateTopic(PublisherClient.FormatTopicName(projectId, topicId));
+            PublisherClient.Create().CreateTopic(new TopicName(projectId, topicId));
 
             // Snippet: CreateSubscriptionAsync(string,string,*,*,CallSettings)
             // Additional: CreateSubscriptionAsync(string,string,*,*,CancellationToken)
```
Comment (chrisdunelm) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs]: string -> SubscriptionName (x2)

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs [hunk 2]
```diff
@@ -219,19 +205,17 @@ public void Pull()
...
-            string topicName = PublisherClient.FormatTopicName(projectId, topicId);
+            TopicName topicName = new TopicName(projectId, topicId);
             publisher.CreateTopic(topicName);
             PubsubMessage newMessage = new PubsubMessage { Data = ByteString.CopyFromUtf8("Simple text") };
-            SubscriberClient.Create().CreateSubscription(SubscriberClient.FormatSubscriptionName(projectId, subscriptionId), topicName, null, 60);
+            SubscriberClient.Create().CreateSubscription(new SubscriptionName(projectId, subscriptionId), topicName, null, 60);
             publisher.Publish(topicName, new[] { newMessage });
 
             // Snippet: PullAsync(string,*,*,CallSettings)
             // Additional: PullAsync(string,*,*,CancellationToken)
```
Comment (chrisdunelm) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs]: string -> SubscriptionName (x2)

**Comment by jskeet** [comment]
Ooh, thanks. I suspect it *will* only be in Pubsub, but clearly needs fixing - will sort that out. Fortunately the AppVeyor build should catch remaining issues...

File: apis/Google.LongRunning/Google.LongRunning.Snippets/OperationsClientSnippets.g.cs [hunk 3]
```diff
@@ -53,9 +53,9 @@ public class GeneratedOperationsClientSnippets
             // Create client
             OperationsClient operationsClient = OperationsClient.Create();
             // Initialize request argument(s)
-            string formattedName = OperationsClient.FormatOperationPathName("[OPERATION_PATH]");
+            string name = "";
             // Make the request
-            Operation response = await operationsClient.GetOperationAsync(formattedName);
+            Operation response = await operationsClient.GetOperationAsync(name);
```
Comment (chrisdunelm) [inline apis/Google.LongRunning/Google.LongRunning.Snippets/OperationsClientSnippets.g.cs]: Has this been auto-gen'ed correctly?
Is "name" meant to no longer be a resource-name?

File: apis/Google.LongRunning/Google.LongRunning/ResourceNames.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Comment (chrisdunelm) [inline apis/Google.LongRunning/Google.LongRunning/ResourceNames.cs]: Is this the correct license?

File: apis/Google.LongRunning/Google.LongRunning.Snippets/OperationsClientSnippets.g.cs (hunk 3 above)
Comment (jskeet) [inline apis/Google.LongRunning/Google.LongRunning.Snippets/OperationsClientSnippets.g.cs]: Given the license issue, it's entirely possible that something odd has happened in terms of autogen. Will look carefully when I get the chance. (I'm not on the right computer to check right now.)

Comment (chrisdunelm) [inline apis/Google.LongRunning/Google.LongRunning.Snippets/OperationsClientSnippets.g.cs]: I think it's here: https://github.com/googleapis/googleapis/blob/master/google/longrunning/longrunning_gapic.yaml#L17
Which specifies the BSD 3-clause license.

Pubsub is defined here: https://github.com/googleapis/googleapis/blob/master/google/pubsub/v1/pubsub_gapic.yaml#L21
Which specifies the Apache 2 license.

File: apis/Google.LongRunning/Google.LongRunning/ResourceNames.cs (hunk 4 above)
Comment (chrisdunelm) [inline apis/Google.LongRunning/Google.LongRunning/ResourceNames.cs]: I think it's here: https://github.com/googleapis/googleapis/blob/master/google/longrunning/longrunning_gapic.yaml#L17
Which specifies the BSD 3-clause license.

Pubsub is defined here: https://github.com/googleapis/googleapis/blob/master/google/pubsub/v1/pubsub_gapic.yaml#L21
Which specifies the Apache 2 license.

File: apis/Google.LongRunning/Google.LongRunning.Snippets/OperationsClientSnippets.g.cs (hunk 3 above)
Comment (jskeet) [inline apis/Google.LongRunning/Google.LongRunning.Snippets/OperationsClientSnippets.g.cs]: The parameter name is just "name" for the method itself. I'm not sure what determines the name of the local variable to use in the snippets.

File: apis/Google.LongRunning/Google.LongRunning/ResourceNames.cs (hunk 4 above)
Comment (jskeet) [inline apis/Google.LongRunning/Google.LongRunning/ResourceNames.cs]: Right. We've got Apache in our hand-written code in LRO, and BSD for the auto-gen already, so this is an existing problem. I suggest we tackle it in a separate PR.

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs (hunk 2 above)
Comment (jskeet) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs]: Done - will leave that as a separate commit, to avoid issues with the LongRunning case-only change getting messed up by a rebase.

**Comment by jskeet** [comment]
Right, I'm going to merge on green:

- We'll sort out the licence issue for LRO in a separate PR
- The name of the variable in snippets is probably something to consider separately
- I've fixed the snippet docs (so AppVeyor should now go green)
